action/protocol/account/transfer.go

```diff
@@ -108,18 +119,24 @@ func (p *Protocol) handleTransfer(ctx context.Context, act action.Action, sm pro
 	if err := sender.SubBalance(tsf.Amount()); err != nil {
 		return nil, errors.Wrapf(err, "failed to update the Balance of sender %s", actionCtx.Caller.String())
 	}
-	// update sender Nonce
-	accountutil.SetNonce(tsf, sender)
+	if fixNonce || tsf.Nonce() != 0 {
+		// update sender Nonce
+		if err := sender.SetPendingNonce(tsf.Nonce() + 1); err != nil {
```

Comment:
I think the condition `fixNonce || tsf.Nonce() != 0` could be moved into SetPendingNonce()

Reply:
how could `fixNonce` be moved into `SetPendingNonce`?